module5/RawParser.py

```diff
@@ -0,0 +1,43 @@
+import os.path
+import pathlib
+from typing import List, Dict
+
+from module4 import string_object_ref as sof
+from module5.InputException import InputException
+
+
+class RawParser:
+    DEFAULT_FILE_PATH = os.path.join(pathlib.Path().resolve(), 'landing', 'raw-publications.txt')
+
+    def __init__(self, file_path):
+        self.file_path = file_path
+
+    def parse_raw_file(self):
+        return self.__get_pub_objects(self.__read_file())
+
+    def __read_file(self) -> List[str]:
+        with open(self.file_path) as file:
+            lines = file.readlines()
+        return lines
+
+    def __get_pub_objects(self, lines: List[str]) -> List[Dict[str, str]]:
+        try:
```

Comment:
Maybe it worth to check more specific exceptions like Wrong number of args for publication or check for some data formats, for example if type ad -> objs[2] should have particular format, etc. But it is not necessary now, I think

Reply:
I agree that the structure of the exceptions in this code is far from perfect :)
In my opinion, building the correct structure is a little out of scope of the program and can take time comparable to an OOP module.